### DISCUSSION TITLE
请问下Buffer::readFd中的 char extrabuf[65536]可以定义成成员变量吗？

### DISCUSSION THREAD

**Discussion by wangdongyu1989** [chenshuo/muduo#602] [Q&A] [answered]
还有 char extrabuf[65536]; 这个长度可以扩大吗 或者有其他考虑？ 多谢

**Comment by chenshuo** [comment, marked as answer]
你如果定义成成员变量，会增加每个 Buffer 的内存开销。
我觉得就失去 stack buffer 的意义了。

如果你的 stack 够大，在 benchmark 证明确有性能提高的前提下，
可以扩大到 128K 或 256K。再大我估计就没啥意义了。
